exchanges/okx/okx_types.go

```diff
@@ -614,13 +640,39 @@ type OrderData struct {
 	SMessage      string `json:"sMsg,omitempty"`
 }
 
+// GetSCode returns a status code value
+func (a *OrderData) GetSCode() string { return a.SCode }
+
+// GetSMsg returns a status message value
+func (a *OrderData) GetSMsg() string { return a.SMessage }
+
+// ResponseSuccess holds responses having a status result value.
+type ResponseSuccess struct {
+	Result bool `json:"result"`
+
+	SCode    string `json:"sCode,omitempty"`
+	SMessage string `json:"sMsg,omitempty"`
+}
+
+// GetSCode returns a status code value
+func (a *ResponseSuccess) GetSCode() string { return a.SCode }
```

Comment:
[Might still be pending](https://github.com/thrasher-corp/gocryptotrader/pull/1420#discussion_r1491803323)

Reply:
These methods are added to be used when we encounter such kind of response with a "code": "1" and the detailed sCode and sMsg are included in the data section.

We can not call the x.SCode and x.SMsg fields in **### handleIncomingData** method. Since we give an input of different instance interfaces as ### dataHolder for the **### handleIncomingData** method.

```
{
  "id": "1514",
  "op": "sprd-cancel-order",
  "data": [
    {
      "clOrdId": "",
      "ordId": "2510789768709120",
      "sCode": "5XXXX",
      "sMsg": "Order not exist"
    }
  ],
  "code": "1",
  "msg": ""
}
```

Reply:
Therefore, removing these _GetSCode()_ and _GetSMsg()_ methods is problematic.